recipes/opentelemetry-proto/all/test_package/conanfile.py

```diff
@@ -1,34 +1,24 @@
-from conan import ConanFile
-from conan.tools.cmake import CMakeDeps, CMakeToolchain
-from conan.tools.env import VirtualBuildEnv
-from conan.tools.layout import basic_layout
 import os
 
+from conan import ConanFile, conan_version
+
+
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
     test_type = "explicit"
 
-    _res_folder = ""
-
     def layout(self):
-        basic_layout(self, src_folder="src")
+        pass
 
     def requirements(self):
         self.requires(self.tested_reference_str)
 
-    def generate(self):
-        tc = CMakeToolchain(self)
-        tc.generate()
-        tc = CMakeDeps(self)
-        tc.generate()
-        tc = VirtualBuildEnv(self)
-        tc.generate(scope="build")
-
-        self._res_folder = self.dependencies["opentelemetry-proto"].conf_info.get("user.opentelemetry-proto:proto_root")
-
     def test(self):
-        # TODO: to remove in conan v2
-        if self._res_folder == "":
-            self._res_folder = self.deps_user_info["opentelemetry-proto"].proto_root
-
-        assert os.path.isfile(os.path.join(self._res_folder, "opentelemetry", "proto", "common", "v1", "common.proto"))
+        if conan_version.major >= 2:
+            res_folder = self.dependencies["opentelemetry-proto"].conf_info.get("user.opentelemetry-proto:proto_root")
+        else:
+            # TODO: to remove in conan v2
+            res_folder = self.deps_user_info["opentelemetry-proto"].proto_root
```

Comment:
Can't this just made to work in both versions with the same v2 approach?

Reply:
@uilianres used this approach for ua-nodeset: https://github.com/conan-io/conan-center-index/commit/5c80fc5d4dbbb719b6e7f743abe3925d5a626f85

Reply:
There is a better approach, you can store it in a file. It's well used by other recipes:


```python
def generate(self):
    save(self, self.dependencies["opentelemetry-proto"].conf_info.get("user.opentelemetry-proto:proto_root"), "proto_root")



def test(self):
    res_folder = load(self, "proto_root")
```